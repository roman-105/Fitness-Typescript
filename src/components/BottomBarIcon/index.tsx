import React from 'react';
import { HomeIcon, ExploreIcon, ProgressIcon, CoachIcon, MyClubIcon } from '../Icon';
import Typography from '../Typography/Typography';
import styles from '../Typography/typography-styles';
import formatMessage from 'format-message';
import { Routes } from '../../router/routes';

const lookup: { [index: string]: (color: string) => JSX.Element } = {
  [Routes.Home]: (color: string) => <HomeIcon fill={color} />,
  [Routes.Programs]: (color: string) => <ExploreIcon fill={color} />,
  [Routes.Progress]: (color: string) => <ProgressIcon fill={color} />,
  [Routes.Coach]: (color: string) => <CoachIcon fill={color} />,
  [Routes.Club]: (color: string) => <MyClubIcon fill={color} />
};

export function bottomBarIcon(name: Routes) {
  return function ({ color }: { color: string }): JSX.Element {
    return lookup[name](color);
  };
}

export function bottomBarLabel(name: Routes) {
  const tabNameLookup: { [index: string]: string } = {
    [Routes.Home]: formatMessage('Home'),
    [Routes.Programs]: formatMessage('Explore'),
    [Routes.Progress]: formatMessage('Progress'),
    [Routes.Coach]: formatMessage('Coach'),
    [Routes.Club]: formatMessage('Clubs')
  };

  return function ({ color }: { color: string }): JSX.Element {
    return (
      <Typography
        style={{ color, ...styles.label }}
        fontFamily="regular"
        fontSize={11}
        lineHeight={12}
      >
        {tabNameLookup[name]}
      </Typography>
    );
  };
}
