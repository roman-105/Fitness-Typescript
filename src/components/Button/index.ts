export { default as BFButton } from './BFButton';
export { default as BFSecondaryButton } from './BFSecondaryButton';
export { default as BFSelectorButton } from './selector/BFSelectorButton';
export { default as BFLightButton } from './BFLightButton';
