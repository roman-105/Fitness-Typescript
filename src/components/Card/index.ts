export { default as MediaCard } from './Media/MediaCard';
export { default as SlashedCard } from './SlashedCard/SlashedCard';
