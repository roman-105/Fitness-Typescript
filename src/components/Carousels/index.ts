export { default as MediaCardCarousel } from './MediaCardCarousel';
import { MediaCardCarouselProps as mediaprops } from './MediaCardCarousel';
export { default as ImageCarousel } from './ImageCarousel';
import { ImageCarouselProps as imageprops } from './ImageCarousel';
export type MediaCardCarouselProps = mediaprops;
export type ImageCarouselProps = imageprops;
