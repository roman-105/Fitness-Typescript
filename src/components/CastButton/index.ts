export { default as BFCastButton } from './BFCastButton';
