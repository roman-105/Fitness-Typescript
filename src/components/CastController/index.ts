export { default as CastController } from './CastController';
