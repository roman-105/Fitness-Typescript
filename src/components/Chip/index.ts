export { default as ChipList } from './ChipList';
export { default as ChipItem } from './ChipItem';
import { Item } from './ChipList';
export type ChipItemType = Item;
