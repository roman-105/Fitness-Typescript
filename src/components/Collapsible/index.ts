export { default as BFCollapsibleView } from './BFCollapsibleView';
