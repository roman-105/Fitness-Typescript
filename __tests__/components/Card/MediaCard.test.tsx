import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { MediaCard } from '../../../src/components/Card';

describe('MediaCard', () => {
  it('renders correctly', async () => {
    const { getByText } = render(<MediaCard name="Media card" onPress={() => {}} />);

    const textElement = getByText('Media card');

    expect(textElement).not.toBeNull();
  });
  it('click on it triggers onPress func', async () => {
    const onPressFn = jest.fn();
    const { getByText } = render(<MediaCard name="Media card" onPress={onPressFn} />);

    const textElement = getByText('Media card');
    fireEvent.press(textElement);

    expect(textElement).not.toBeNull();
    expect(onPressFn).toHaveBeenCalledTimes(1);
  });

  it('snapshot', async () => {
    const { toJSON } = render(<MediaCard name="Media card" onPress={() => {}} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
