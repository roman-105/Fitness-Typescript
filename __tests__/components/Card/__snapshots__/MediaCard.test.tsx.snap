// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaCard snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 2,
        "elevation": 1,
        "maxHeight": 300,
        "shadowColor": "rgba(0,0,0,0.16)",
        "shadowOffset": Object {
          "height": 4,
          "width": 4,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1,
        "width": "100%",
      },
      Array [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": "85%",
        "justifyContent": "space-between",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "justifyContent": "flex-end",
              "marginBottom": -1,
              "overflow": "hidden",
              "width": "100%",
            },
            Object {
              "height": "80%",
            },
            Object {
              "borderBottomColor": "#FFFFFF",
              "justifyContent": "space-between",
            },
          ],
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={null}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View />
      <View
        style={
          Object {
            "borderBottomColor": "#FFFFFF",
            "borderBottomWidth": 25,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 400,
            "borderStyle": "solid",
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 16,
          "paddingTop": 6,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "maxWidth": "100%",
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "Impact",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              undefined,
              undefined,
            ]
          }
        >
          Media card
        </Text>
      </View>
    </View>
  </View>
</View>
`;
